scripts/caclmgrd

```diff
@@ -890,6 +894,17 @@ class ControlPlaneAclManager(daemon_base.DaemonBase):
                         self.num_changes[namespace] = 0
                         self.update_thread[namespace] = None
                         return
+        except Exception as e:
+            self.log_error("Exception occured at {} thread due to {}".format(threading.current_thread().getName(), repr(e)))
+            exc_type, exc_value, exc_traceback = sys.exc_info()
+            msg = traceback.format_exception(exc_type, exc_value, exc_traceback)
+            for tb_line in msg:
+                for tb_line_split in tb_line.splitlines():
+                    self.log_error(tb_line_split)
```

Comment:
>log_error [](http://example.com/codeflow?start=25&length=9)

syslog may drop message or mess order due to UDP protocol. suggest queue the exception, and re-raise in main thread.

Alternatively, using join in main thread to handle thread exception. No need to invent exception_queue.

Reply:
In main thread, it still needs to keep checking if there are next db updates, so it can't join and wait for the child thread's result and handle the exception, that's why I choose exception queue and only check if the queue is empty or not before starting checking the db updates every time in main thread.